plugin/advisor/mysql/advisor_naming_column.go

```diff
@@ -0,0 +1,132 @@
+	"github.com/bytebase/bytebase/plugin/db"
+	"github.com/pingcap/tidb/parser/ast"
+)
+
+var (
+	_ advisor.Advisor = (*NamingColumnConventionAdvisor)(nil)
+)
+
+func init() {
+	advisor.Register(db.MySQL, advisor.MySQLNamingColumnConvention, &NamingColumnConventionAdvisor{})
+	advisor.Register(db.TiDB, advisor.MySQLNamingColumnConvention, &NamingColumnConventionAdvisor{})
+}
+
+// NamingColumnConventionAdvisor is the advisor checking for column naming convention.
+type NamingColumnConventionAdvisor struct {
+}
+
+// Check checks for column naming convention.
+func (adv *NamingColumnConventionAdvisor) Check(ctx advisor.Context, statement string) ([]advisor.Advice, error) {
+	root, errAdvice := parseStatement(statement, ctx.Charset, ctx.Collation)
+	if errAdvice != nil {
+		return errAdvice, nil
+	}
+
+	level, err := advisor.NewStatusBySchemaReviewRuleLevel(ctx.Rule.Level)
+	if err != nil {
+		return nil, err
+	}
+	format, err := api.UnmarshalNamingRulePayloadFormat(ctx.Rule.Payload)
+	if err != nil {
+		return nil, err
+	}
+	checker := &namingColumnConventionChecker{
+		level:  level,
+		format: format,
+		tables: make(tableState),
+	}
+
+	for _, stmtNode := range root {
+		(stmtNode).Accept(checker)
+	}
+
+	return checker.generateAdviceList(), nil
+}
+
+type namingColumnConventionChecker struct {
+	adviceList []advisor.Advice
+	level      advisor.Status
+	format     *regexp.Regexp
+	tables     tableState
+}
+
+func (v *namingColumnConventionChecker) Enter(in ast.Node) (ast.Node, bool) {
+	switch node := in.(type) {
+	// CREATE TABLE
+	case *ast.CreateTableStmt:
+		v.createTable(node)
+	// ALTER TABLE
+	case *ast.AlterTableStmt:
+		table := v.tables.getTable(node.Table.Name.O)
+		for _, spec := range node.Specs {
+			switch spec.Tp {
+			// RENAME COLUMN
+			case ast.AlterTableRenameColumn:
+				delete(table, spec.OldColumnName.Name.O)
+				table[spec.NewColumnName.Name.O] = true
+			// ADD COLUMNS
+			case ast.AlterTableAddColumns:
+				for _, column := range spec.NewColumns {
+					table[column.Name.Name.O] = true
+				}
+			// DROP COLUMN
+			case ast.AlterTableDropColumn:
+				delete(table, spec.OldColumnName.Name.O)
+			// CHANGE COLUMN
+			case ast.AlterTableChangeColumn:
+				delete(table, spec.OldColumnName.Name.O)
+				table[spec.NewColumns[0].Name.Name.O] = true
+			}
+		}
+	}
+	return in, false
+}
+
+func (v *namingColumnConventionChecker) Leave(in ast.Node) (ast.Node, bool) {
+	return in, true
+}
+
+func (v *namingColumnConventionChecker) createTable(node *ast.CreateTableStmt) {
+	table := make(columnSet)
+	for _, column := range node.Cols {
+		table[column.Name.Name.O] = true
+	}
+	v.tables[node.Table.Name.O] = table
+}
+
+func (v *namingColumnConventionChecker) generateAdviceList() []advisor.Advice {
+	tableList := v.tables.tableList()
+	for _, tableName := range tableList {
+		table := v.tables[tableName]
+		columnList := table.columnList()
+		for _, columnName := range columnList {
+			if !v.format.MatchString(columnName) {
```

Comment:
I find in the [`advisor_naming_table.go`](https://github.com/bytebase/bytebase/blob/ea160fcf63f6d967d3c6030d1378d8909438b657/plugin/advisor/mysql/advisor_naming_table.go#L94) , we will break the loop as long as the name mismatch. Do we need to keep the same logic between these two check?

Reply:
Thanks. I'll modify the logic in `advisor_naming_table.go` later(another PR).